crates/gitql-ast/src/function.rs

```diff
@@ -242,6 +250,26 @@ fn text_char(inputs: Vec<Value>) -> Value {
     return Value::Text("".to_string());
 }
 
+fn text_stuff(inputs: Vec<Value>) -> Value {
+    let text = inputs[0].as_text();
+    let start = (inputs[1].as_number() - 1) as usize;
+    let length = inputs[2].as_number() as usize;
+    let new_string = inputs[3].as_text();
+
+    if text.is_empty() {
+        return Value::Text(text);
+    }
+    if start as usize > text.len() || length as usize > text.len() {
+        return Value::Text(text);
+    }
+
+    let mut text = text.chars().collect::<Vec<_>>();
+    let new_string = new_string.chars().collect::<Vec<_>>();
+    text.splice(start as usize..(start + length), new_string);
+
```

Comment:
Looks good for me but we need to double check if std has function for this purpose

Reply:
I don't think so, I checked before implementing it